### PR TITLE
AT_001.022 | New item > Create Freestyle project

### PR DESCRIPTION
**Pull request by misssEvil**
# TC_001.038 New item > Create Freestyle project 

**Pre-conditions:** run chrome driver and log in into Jenkins
**Steps:**
* click on "new item" and redirect to creating page
* input parameter ProjectName into name field
* click OK button
* do to dashboard
* find on jobs table needed item and click

**Parameters:**
1. ProjectName = newFreestyleProjectValidName
2. ProjectName = newFreestyleProjectInvalidName[]//*
3. ProjectName is null
4. ProjectName = new Freestyle Project

**Expected result:** current URL is a detail page of created item

**Post-conditions:** delete created test item, log out and quite

### CODE REVIEW

**Comment by misssEvil** [comment]
[AT_001.022 | New item > Create Freestyle project](https://trello.com/c/CkOYqzQD/658-at001022-new-item-create-freestyle-project)

**Review by li-ma1 — APPROVED** [review]

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        Assert.assertEquals(getDriver().getCurrentUrl(), "http://localhost:8080/job/newFreestyleProject/");
+        deleteFreestyleProject(projectBName);
+    }
+
+    @Test (description = "TC_001.038 New item > Create Freestyle project ")
+    public void createFreestyleProjectWithInvalidName() {
+        String projectBName = "[]//*";
+        getDriver().findElement(By.linkText("New Item")).click();
+        getDriver().findElement(By.id("name")).sendKeys(projectBName);
+        getDriver().manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
```
Comment (RusAli) [inline src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:49]: don't use implicitlywait in this case. BaseUtils has already implemented this waiter in createDriver() method.
Please watch video by Sergey about waiters.

U may use getwait5() or getwait20() methods

**Review by EugeniaSrednitskaya — APPROVED** [review]

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        deleteFreestyleProject(projectBName);
+    }
+
+    @Test (description = "TC_001.038 New item > Create Freestyle project ")
+    public void createFreestyleProjectNullName()  {
+        String projectBName = "lalala";
+        getDriver().findElement(By.linkText("New Item")).click();
+        getDriver().findElement(By.id("name")).sendKeys(projectBName);
+        getDriver().findElement(By.id("name")).clear();
+        getDriver().manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
```
Comment (RusAli) [inline src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:72]: The same point. Please use another waiter

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    @Test (description = "TC_001.038 New item > Create Freestyle project ")
+    public void createFreestyleProjectWithValidName() {
+        String projectBName = "newFreestyleProject";
+        getDriver().findElement(By.linkText("New Item")).click();
+        getDriver().findElement(By.id("name")).sendKeys(projectBName);
+        choseNewItemType(NEW_JOB_FREESTYLE_CLASS);
+        getDriver().findElement(By.id("ok-button")).click();
+        goToDashBoard();
+        findItemOnJobsTableAndClick(projectBName);
+        Assert.assertEquals(getDriver().getCurrentUrl(), "http://localhost:8080/job/newFreestyleProject/");
```
Comment (RusAli) [inline src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:40]: Please divide assertion with white space between logical steps.

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    @Test (description = "TC_001.038 New item > Create Freestyle project ")
+    public void createFreestyleProjectNullName()  {
+        String projectBName = "lalala";
+        getDriver().findElement(By.linkText("New Item")).click();
+        getDriver().findElement(By.id("name")).sendKeys(projectBName);
+        getDriver().findElement(By.id("name")).clear();
+        getDriver().manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
+        Assert.assertTrue(getDriver().findElement(By.id("itemname-required")).isEnabled());
+    }
+}
```
Comment (RusAli) [inline src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:75]: So one more point. One Test - one PR

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java (hunk 1 above)
Comment (misssEvil) [reply src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:49]: Thank you for your comment!
Using implicit wait is not a good practice, my bad. Changed it into getwait() method

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java (hunk 3 above)
Comment (misssEvil) [reply src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:40]: Fair point! Thanks!

File: src/test/java/CreateNewFreestyleProjectTest_TC001_038.java (hunk 4 above)
Comment (misssEvil) [reply src/test/java/CreateNewFreestyleProjectTest_TC001_038.java:75]: It's one test case, running a few times with different parameters, so I've intentionally decided to put them into one file.

**Pull request merged**